### PR TITLE
Made all errors specific in stmt.rs

### PR DESCRIPTION
**Pull request by pythondude325**

### CODE REVIEW

File: src/ir/stmt.rs [hunk 1]
```diff
@@ -357,16 +339,10 @@ impl Compiler {
...
-                        "'{}' statement not in loop or switch statement",
-                        if is_break { "break" } else { "continue" }
-                    ),
-                    location
-                );
+                Err(location.error(SemanticError::LoopExitOutsideLoopOrSwitch { is_break }))
             }
         } else if !is_break {
-            semantic_err!("'continue' not in loop".into(), location);
+            Err(location.error(SemanticError::ContinueNotInLoop))
```
Comment (pythondude325) [inline src/ir/stmt.rs]: I can combine these two errors if you want

File: src/data/error.rs [hunk 2]
```diff
@@ -114,6 +114,27 @@ pub enum SemanticError {
...
+    LabelRedeclaration(cranelift::prelude::Ebb),
+
+    #[error("use of undeclared label {0}")]
+    UndeclaredLabel(crate::intern::InternedStr),
+
+    #[error("{}case outside of switch statement", if *(.is_default) { "default " } else { "" })]
+    CaseOutsideSwitch { is_default: bool },
+
+    #[error("cannot have multiple default cases in a switch statement")]
+    MultipleDefaultCase,
```
Comment (jyn514) [inline src/data/error.rs]: This is part of a more general error which is "case repeated in switch statement". We don't catch that one currently but we will soon: https://github.com/jyn514/rcc/issues/221

Comment (pythondude325) [reply src/data/error.rs]: We can change the error once the other duplicate cases are detected

Comment (jyn514) [reply src/data/error.rs]: Can you change the name to DuplicateCase? It looks good after that.

Comment (jyn514) [reply src/data/error.rs]: I'll do this myself when I fix https://github.com/jyn514/rcc/issues/221

**Pull request merged**